src/vmcs/src/ept_intel_x64.cpp

```diff
@@ -42,36 +42,50 @@ ept_entry_intel_x64
 ept_intel_x64::add_page(integer_pointer gpa, integer_pointer bits, integer_pointer end)
 {
     auto &&index = ept::index(gpa, bits);
+    auto &&view = gsl::make_span(m_ept, ept::num_entries);
 
     if (bits > end)
     {
+        auto &&entry = ept_entry_intel_x64(&view.at(index));
+        if (entry.entry_type())
```

Comment:
This checks if the entry maps a page..the m_ept stores both the page and table entries